src/main/java/uk/gov/di/ipv/cri/passport/service/PassportService.java

```diff
@@ -5,79 +5,59 @@
-    private final DcsSigningService dcsSigningService;
-    private final DcsEncryptionService dcsEncryptionService;
-    private final DataStore<DcsResponseItem> dataStore;
+    private final DataStore<PassportCheckDao> dataStore;
     private final HttpClient httpClient;
 
     public PassportService(
             HttpClient httpClient,
             ConfigurationService configurationService,
-            DcsEncryptionService dcsEncryptionService,
-            DcsSigningService dcsSigningService,
-            DataStore<DcsResponseItem> dataStore) {
+            DataStore<PassportCheckDao> dataStore) {
         this.httpClient = httpClient;
         this.configurationService = configurationService;
-        this.dcsEncryptionService = dcsEncryptionService;
-        this.dcsSigningService = dcsSigningService;
         this.dataStore = dataStore;
     }
 
     public PassportService()
             throws CertificateException, NoSuchAlgorithmException, InvalidKeySpecException,
                     KeyStoreException, IOException {
         this.configurationService = new ConfigurationService();
-        this.dcsSigningService = new DcsSigningService();
-        this.dcsEncryptionService = new DcsEncryptionService();
         this.dataStore =
                 new DataStore<>(
                         configurationService.getDcsResponseTableName(),
-                        DcsResponseItem.class,
-                        DataStore.getClient());
+                        PassportCheckDao.class,
+                        DataStore.getClient(configurationService.getDynamoDbUri()));
         this.httpClient = HttpClientSetUp.generateHttpClient(configurationService);
     }
 
-    public String dcsPassportCheck(String payload) throws IOException {
-        try {
-            JWSObject signedPayload = dcsSigningService.signData(payload);
-            String encryptedPayload = dcsEncryptionService.encrypt(signedPayload.serialize());
-            String reSignedPayload = dcsSigningService.signData(encryptedPayload).serialize();
+    public DcsSignedEncryptedResponse dcsPassportCheck(JWSObject payload) throws IOException {
+        HttpPost request = new HttpPost(configurationService.getDCSPostUrl());
+        request.addHeader("content-type", "application/jose");
+        request.setEntity(new StringEntity(payload.serialize()));
 
-            var request = new HttpPost(configurationService.getDCSPostUrl());
-            request.addHeader("content-type", "application/jose");
-            request.setEntity(new StringEntity(reSignedPayload));
+        HttpResponse response = httpClient.execute(request);
 
-            HttpResponse response = httpClient.execute(request);
-
-            if ((response != null)) {
-                return response.toString();
-            }
-        } catch (Exception e) {
-            e.printStackTrace();
+        if (response == null) {
+            throw new NullPointerException("Response from DCS is null");
```

Comment:
Should the response we wrapped as an optional, and then handle it rather than throwing a NPE? I don't think the NPE is being caught in the PassportHandler - would be good to handle it and return a 500 gracefully.